pages/posts/2025-02-26-how-i-approach-code-review.md

```diff
@@ -0,0 +1,101 @@
+    - Another example: are we waiting for another team like our privacy counsel, legal administrators, or security engineers to "clear" that aspect of the changes before this can go out?
+- Automated checks
+  - Does the continuous integration (CI) checks pass? If not, what checks are pending, timed out, failed, or errored, and is the author already aware of those failures?
+
+## Reviewing the diff
+
+Before I start my review of the code changes, I look at what others have done already to further optimize my personal context:
+
+- What have previous reviewers already commented on?
+- If AI tools have been enabled (like Copilot), what have they commented on?
+
+With that in mind (or if starting on an "evergreen" yet-to-be reviewed pull request), I go through these questions:
+
+- How many files have changed?
+- What kinds of files have changed?
+  - Do they all "belong" in this pull request, or are there specific files that can (or should) land separately in their own pull requests?
+  - This is important in "bigger" codebases like GitHub.com, where we prioritize availability and the cost of landing a bigger set of changes can have a downstream impact on other teams and more importantly our customers who rely on us to be fully available for their work.
+  - How many files are user-facing? What about the files that are service or system-facing only to developers? What's that ratio look like?
+- How are the files changed connected to each other? Are there any "pork-barrel" changes that are nice-to-have but not-exactly-in-scope based on the `Gathering context` stage that should be called out?
+
+In my review, I use the "Start a review" feature to make a series of comments before submitting the final review (requesting changes, approved, or commenting). In each comment, I think of the following types and preface my comment with one of these expressions to set expectations for what I'd like the authors to be aware of.
+
+- `💅 Non-blocking suggestion`: if there's formatting that would be otherwise better handled by a developer tool (linter; bot; script) or a rewording, I find these helpful for the author to consider as a potential improvement without it blocking my approval for the changeset as a whole.
+- `🙋 Question`: When I don't understand something or need more context about its relevance. I try really hard _not_ to sneak questions in as a requested change or a suggestion, but to draw out more from the author's intentions before deciding on what to do next.
+- `🟡 Requested change`: when I know this area must either be reviewed and addressed (be it a change, correction, or a defense against change) before I can personally stand behind the approval of the changeset as a whole.
+- `✨ Affirmation`: when I see something that I see is well-done or well-expressed. Can be anything from documenting a previously undocumented method or a way of improving performance, accessibility, or security in some way!
```

Comment:
I'd love for us to use these as part of our NUX dev standards: https://github.slack.com/archives/C084P2BJNQ7/p1740672369700289

Reply:
Absolutely! I see that a variation has already been published, so I'll definitely make more edits there. Thank you!